addons/survey/static/src/js/survey_form.js

```diff
@@ -168,10 +191,19 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend(SurveyPreloa
             $commentInput.closest('.o_survey_comment_container').addClass('d-none');
             $commentInput.each((idx, $input) => $input.disabled = true);
         }
-
-        var $matrixBtn = $target.closest('.o_survey_matrix_btn');
+    },
+    /**
+     * For single and multiple choice questions, propagate questions visibility
+     * based on conditional questions and (de)selected triggers
+     *
+     * @param {JQuery<HTMLElement>} $target
+     * @param {JQuery<HTMLElement>} $choiceItemGroup
+     * @returns {boolean} Whether the question is considered completed
+     */
+    _applyAnswerSelectionChanges: function ($target, $choiceItemGroup) {
```

Comment:
If it now strictly checks conditional questions visibility, maybe "_toggleConditionalQuestionsVisibility"